test/e2e/tls/tls_test.go

```diff
@@ -77,19 +77,45 @@ allow {
 
 	// We need a client with proper TLS setup, otherwise the health check
 	// that loops to determine if the server is ready will fail.
-	testRuntime.Client = newClient(pool, "testdata/client-cert.pem", "testdata/client-key.pem")
+	testRuntime.Client = newClient(0, pool, "testdata/client-cert.pem", "testdata/client-key.pem")
 
 	os.Exit(testRuntime.RunTests(m))
 }
 
+func TestMinTLSVersion(t *testing.T) {
+	endpoint := testRuntime.URL()
+	t.Run("TLS version not suported by server", func(t *testing.T) {
+
+		c := newClient(tls.VersionTLS10, pool, "testdata/client-cert.pem", "testdata/client-key.pem")
+		_, err := c.Get(endpoint)
+		if err == nil {
+			t.Error("expected err - protocol version not supported, got nil")
+		}
+
+	})
+	t.Run("TLS Version supported by server", func(t *testing.T) {
+
+		c := newClient(tls.VersionTLS12, pool, "testdata/client-cert.pem", "testdata/client-key.pem")
+		resp, err := c.Get(endpoint)
+		if err != nil {
+			t.Fatalf("GET: %v", err)
+		}
+		defer resp.Body.Close()
+		if resp.StatusCode != http.StatusOK {
+			t.Errorf("expected status 200, got %s", resp.Status)
+		}
+	})
+
```

Comment:
❓ Would it make sense to add a case where the server's min version is different from the default, say, 1.3?

Reply:
Done